torchft/process_group.py

```diff
@@ -1012,6 +1118,56 @@ def allgather(
 
         return self._run_func("allgather", output_tensors, input_tensor, opts)
 
+    def allreduce(
+        self,
+        tensors: List[torch.Tensor],
+        opts: Union[dist.AllreduceOptions, dist.ReduceOp],
+    ) -> Work:
+        assert isinstance(tensors, list), "input must be list"
+
+        for tensor in tensors:
```

Comment:
might be worth adding helper methods for common patterns like 
`maybe_share_tensors(tensor_or_tensors)`

`assert_list(tensors)`
etc?

Reply:
yea that's clean, added